backend/app/usecase/search/order/created_time.go

```diff
@@ -0,0 +1,43 @@
+package order
+
+import (
+	"sort"
+
+	"github.com/short-d/short/backend/app/entity"
+)
+
+var _ Order = (*CreatedTime)(nil)
+
+// CreatedTime arranges searchable resources based on their created time.
+type CreatedTime struct {
+}
+
+// ArrangeShortLinks arranges shortLinks based on CreatedAt.
+// The returned short links satisfies the created before comparision if CreatedAt
+// is not nil, otherwise it sits on the end of the slices if CreatedAt is nil.
+func (c CreatedTime) ArrangeShortLinks(shortLinks []entity.ShortLink) []entity.ShortLink {
+	sort.SliceStable(shortLinks, func(i, j int) bool {
+		if shortLinks[j].CreatedAt == nil {
+			return true
+		} else if shortLinks[i].CreatedAt == nil {
+			return false
+		}
+		return shortLinks[i].CreatedAt.Before(*shortLinks[j].CreatedAt)
```

Comment:
```suggestion
		iif first := shortLinks[firstIdx].CreatedAt
                 second := shortLinks[secondIdx].CreatedAt
		if first == nil && first == nil {
			return true
		}
		
		if first == nil {
		    return false
		}
		
		if second == nil {
		    return true
		}
		return first.Before(second)
```

Reply:
I agree that your suggestion is easier to read. I will modify it.